web/src/api.ts

```diff
@@ -178,23 +198,36 @@ export const deleteCommit = async (commitId: number, idToken: string) => {
 };
 
 /**
- * Retrieves merchant with the given email from the database.
+ * Retrieves all merchants from the database.
+ * @param filters Filters used to restrict merchants retrieved
  */
-export const getMerchantWithEmail = async (
-  email: string
-): Promise<MerchantResponse> => {
-  const res = await fetch(
-    `${process.env.REACT_APP_SERVER_URL}/merchants?email=${email}`
-  );
+const getAllMerchants = async (queryParams?: {
+  [key: string]: string;
+}): Promise<MerchantResponse[]> => {
+  const endpoint = `${process.env.REACT_APP_SERVER_URL}/merchants`;
+  const res = queryParams
+    ? await query(endpoint, queryParams)
+    : await fetch(endpoint);
 
   if (res.status !== 200) {
     throw new Error(GENERIC_ERROR);
   }
 
-  const merchants = await res.json();
+  return res.json();
+};
+
+/**
+ * Retrieves merchant with the given Firebase UID from the database.
+ */
+export const getMerchantWithFirebaseUid = async (
+  firebaseUid: string
+): Promise<MerchantResponse> => {
+  const merchants = await getAllMerchants({
+    firebaseUid,
+  });
 
   if (merchants.length === 0) {
-    throw new Error(USER_NOT_FOUND);
+    throw new Error(NO_MERCHANT_WITH_FIREBASE_UID);
```

Comment:
Include the Firebase UID that is throwing error in this error message as well.

Reply:
@xlai20 okay!